posts/kubernetes-tutorial-1-intro.md

```diff
@@ -0,0 +1,69 @@
+
+Kubernetes is an open source project on [github](https://github.com/kubernetes/kubernetes). They define their project as: 
+> Kubernetes, also known as K8s, is an open source system for managing containerized applications across multiple hosts. It provides basic mechanisms for deployment, maintenance, and scaling of applications.
+
+## Application Deployment
+
+To understand K8s, I believe it's helpful to look at and compare how we deployed production applications in the past and what k8s brings to the table. First lets consider the our first old school setup. At company ABC there is a couple of servers in the basement for hosting our applications. Lets say we size our production servers to match our expected user base and concurrent usage. In our example we have four business applications (Green,Orange,Teal,Blue). Lets say all these applications share a network, database and load balancer with each other. We also have a couple of spare servers allocated in the event of a disaster recovery.
+
+![onprem-deployment](https://github.com/David-Elizondo/blog-usa/blob/master/images/2021/09/onprem.png)
+
+## Simulating a Disaster
+
+Lets say the IT get alerted about an outage of application Blue. An electrical storm has come by that night and shorted the hardware on the server. During the outage IT determines that they will need to move the application to one of the spare servers. 
+
+![onprem-deployment](https://github.com/David-Elizondo/blog-usa/blob/master/images/2021/09/onprem2.png)
+
+During the outage IT will be scrambling to bring one of the spare servers online. Configure access the shared database, network load balancer and allowing firewall rules for the application to run on the new server. Each of these task can be time consuming if not planned for. 
+
+In our example each application can not recover from a single machine failure. Recovery from said disaster will bring outage times and IT time from investigation to recovery. Naturally we'd ask how can be setup our environment such that our application can recover from this scenario?
+
+## Break Down 
+
+In our example losing one machine means having to manually recover. There are many ways to make your recovery time better in this scenario. For instance more equipment to protect against electrical storm or establish a process to build and test a DR plan in case of such disaster. These type of protections don't change the underlying issue which is single machine recovery. Another method used on-prem is to have one hot machine and a cold machine. The hot machine is used until a issue arises where the cold machine to switched to in case of backup. In this way now you can recover from a single failure but you've only moved the bar one tick and still would not be able to recover from a two machine failure. Ideally we wan't to be able to recovery from N machine failures. 
```

Comment:
```suggestion
In our example, losing one machine means having to manually recover. There are many ways to make your recovery time faster in this scenario. For instance, more equipment to protect against electrical storm or establish a process to build and test a DR plan in case of such disaster. These types of protections don't change the underlying issue which is single machine recovery. Another method used on-premises is to have one hot machine and a cold machine. The hot machine is used until an issue arises where the cold machine to switched to in case of backup. In this way now you can recover from a single failure, but you've only moved the bar one tick and still would not be able to recover from a two-machine failure. Ideally, we won't to be able to recover from an arbitrarily large amount of machine failures. 
```